src/pages/landing-page/landing-page.jsx

```diff
@@ -1,30 +1,56 @@
-import React from 'react'
+import React, {useState,useEffect} from 'react'
 import './landing-page.css'
 import Banner from '../../assets/mobile-banner.jpg'
 import { Categories } from '../../components/categories/categories'
 import {Card} from '../../components/card/card'
+import { useProduct } from '../../contexts/product-data-context'
+import axios from "axios";
+import { Link } from 'react-router-dom'
+
+
 
 const LandingPage = () => {
+
+  const {data}=useProduct();
+  const topProducts= data.filter((item)=>item.topProduct)
+
+  const[category,setCategory]=useState([])
+
+    useEffect(()=>{
+    const fetch=async()=>{
```

Comment:
This looks fine would suggest you to name it as fetchData, fetchCategory that would be more descriptive